modules/cdc-ext/src/test/java/org/apache/ignite/cdc/AbstractReplicationTest.java

```diff
@@ -0,0 +1,331 @@
+
+    /** @return Test parameters. */
+    @Parameterized.Parameters(name = "cacheMode={0},backupCnt={1}")
+    public static Collection<?> parameters() {
+        List<Object[]> params = new ArrayList<>();
+
+        for (CacheAtomicityMode mode : EnumSet.of(ATOMIC, TRANSACTIONAL))
+            for (int i = 0; i < 2; i++)
+                params.add(new Object[] {mode, i});
+
+        return params;
+    }
+
+    /** */
+    public static final String AP_CACHE = "active-passive-cache";
+
+    /** */
+    public static final String ACTIVE_ACTIVE_CACHE = "active-active-cache";
+
+    /** */
+    public static final byte SRC_CLUSTER_ID = 26;
+
+    /** */
+    public static final byte DEST_CLUSTER_ID = 27;
+
+    /** */
+    public static final int EXISTS = 1;
+
+    /** */
+    public static final int REMOVED = 2;
+
+    /** */
+    public static final int KEYS_CNT = 50;
+
+    /** */
+    protected static IgniteEx[] srcCluster;
+
+    /** */
+    protected static IgniteConfiguration[] srcClusterCliCfg;
+
+    /** */
+    protected static IgniteEx[] destCluster;
+
+    /** */
+    protected static IgniteConfiguration[] destClusterCliCfg;
+
+    /** */
+    private int commPort = TcpCommunicationSpi.DFLT_PORT;
+
+    /** */
+    private int discoPort = TcpDiscoverySpi.DFLT_PORT;
+
+    /** */
+    private byte clusterId = SRC_CLUSTER_ID;
+
+    /** */
+    protected int clientsCnt = 1;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName)
+            .setDiscoverySpi(new TcpDiscoverySpi()
+                .setLocalPort(discoPort)
+                .setIpFinder(new TcpDiscoveryVmIpFinder() {{
+                    setAddresses(Collections.singleton("127.0.0.1:" + discoPort + ".." + (discoPort + DFLT_PORT_RANGE)));
+                }}))
+            .setCommunicationSpi(new TcpCommunicationSpi()
+                .setLocalPort(commPort));
+
+        if (!cfg.isClientMode()) {
+            CacheVersionConflictResolverPluginProvider<?> cfgPlugin = new CacheVersionConflictResolverPluginProvider<>();
+
+            cfgPlugin.setClusterId(clusterId);
+            cfgPlugin.setCaches(new HashSet<>(Collections.singletonList(ACTIVE_ACTIVE_CACHE)));
+            cfgPlugin.setConflictResolveField("reqId");
+
+            cfg.setPluginProviders(cfgPlugin);
+
+            cfg.setDataStorageConfiguration(new DataStorageConfiguration()
+                .setDefaultDataRegionConfiguration(new DataRegionConfiguration()
+                    .setPersistenceEnabled(true)));
+
+            cfg.getDataStorageConfiguration()
+                .setWalForceArchiveTimeout(5_000)
+                .setChangeDataCaptureEnabled(true);
+
+            cfg.setConsistentId(igniteInstanceName);
+        }
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
```

Comment:
method content describes 2 pretty similar operations (cluster configurations). 
could we deduplicate this?